### PR TITLE
fix: in case the location header has slash at the beginning

### PR DESCRIPTION
**Pull request by AmirthavalliG**
When location header is of following "/Patient/3ac58844-5c7e-4b4f-8600-c0b836bfa48a/_history/f52ecc0d-16ef-4f5a-9183-9306f103685b"  the getBindingInfo fails throwing an error from model since extra / is getting appended.

TODO: Have a normalised function a central place to do this

### CODE REVIEW

**Comment by AmirthavalliG** [comment]
@flovogt : Added the tests cases

File: test/qunit/model/FHIRModel.unit.js [hunk 1]
```diff
@@ -1143,4 +1144,27 @@ sap.ui.define([
...
+			name: [
+				{
+					family: "james",
+					given: ["parker"]
+				}
+			]
+		});
+		var sPatientPath = "/Patient/" + sResourceId;
+		this.oFhirModel1.bindContext(sPatientPath);
+		var oResource = this.oFhirModel1.oData.Patient[sResourceId];
```
Comment (flovogt) [inline test/qunit/model/FHIRModel.unit.js:1159]: use `getProperty` function

File: test/qunit/model/FHIRModel.unit.js [hunk 2]
```diff
@@ -1143,4 +1144,27 @@ sap.ui.define([
...
+		var sFullUrl = "urn:uuid:" + sResourceId;
+		var mResponseHeaders = { "etag": "W/\"1\"", "location": sPatientPath + "/_history/1" };
+		var oBundleEntry = new FHIRBundleEntry(sFullUrl, oResource, undefined);
+		var oUpdatedResource = this.oFhirModel1._getUpdatedResourceFromFHIRResponse(mResponseHeaders, oBundleEntry);
+		assert.strictEqual(oUpdatedResource.id, sResourceId);
```
Comment (flovogt) [inline test/qunit/model/FHIRModel.unit.js:1164]: give a message to the assertion

File: test/qunit/model/FHIRModel.unit.js [hunk 3]
```diff
@@ -1143,4 +1144,27 @@ sap.ui.define([
...
+		mResponseHeaders = { "etag": "W/\"1\"", "location": "Patient/" + sResourceId + "/_history/1" };
+		oUpdatedResource = this.oFhirModel1._getUpdatedResourceFromFHIRResponse(mResponseHeaders, oBundleEntry);
+		assert.strictEqual(oResource.id, sResourceId);
```
Comment (flovogt) [inline test/qunit/model/FHIRModel.unit.js:1167]: give a message to the assertion

File: test/qunit/model/FHIRModel.unit.js (hunk 1 above)
Comment (AmirthavalliG) [reply test/qunit/model/FHIRModel.unit.js]: done

File: test/qunit/model/FHIRModel.unit.js (hunk 3 above)
Comment (AmirthavalliG) [reply test/qunit/model/FHIRModel.unit.js]: done

File: test/qunit/model/FHIRModel.unit.js (hunk 2 above)
Comment (AmirthavalliG) [reply test/qunit/model/FHIRModel.unit.js]: done

File: test/qunit/model/FHIRModel.unit.js [hunk 4]
```diff
@@ -1143,4 +1144,27 @@ sap.ui.define([
...
+		var sResourceId = this.oFhirModel1.create("Patient", {
+			name: [
+				{
+					family: "james",
+					given: ["parker"]
+				}
+			]
+		});
+		var sPatientPath = "/Patient/" + sResourceId;
+		this.oFhirModel1.bindContext(sPatientPath);
```
Comment (flovogt) [inline test/qunit/model/FHIRModel.unit.js:1158]: Why we need the call of bindContext here? In general, testing private methods its not intended. 
The test scenario 1 could look like following:
1. Create Model with dummy data
2. Simulate GET request and return location header without slash
3. Test if resource is stored correctly

The test scenario 2 could look like following:
1. Create Model with dummy data
2. Simulate GET request and return location header with slash
3. Test if resource is stored correctly

Comment (AmirthavalliG) [reply test/qunit/model/FHIRModel.unit.js]: @flovogt : i have tried approach as suggested. let me know if its fine..

Comment (AmirthavalliG) [reply test/qunit/model/FHIRModel.unit.js]: @flovogt In one of the response entry I have added / in location and tested without the fix and the test cases are failing

**Pull request merged**